JetUtilities/src/L2Creator.cc

```diff
@@ -431,372 +550,374 @@ bool L2Creator::getL3Rsp() {
+  //
+  // set the best parameters and chi2 to the fit function
+  //
+  TF1 * ffh = gabscor->GetFunction("fit");
+  for (unsigned int np=0;np < bestPars.size() ; np++){
+    ffh->SetParameter(np,bestPars[np]);
+    fabscor->SetParameter(np,bestPars[np]);
+  }
+  fabscor->SetChisquare(bestRChi2 * fabscor->GetNDF());
+  ffh->SetChisquare(bestRChi2 * fabscor->GetNDF());
+
+  //
+  // warn if the fit diverges at low pt
+  //
+  if (fabscor->Integral(0,10) > 60)
+    cout << "\t***ERROR***, fit for histo " << gabscor->GetName() << " diverges at low pt" << endl;
+
+  //   
+  // check for failed fits
+  // a chi2 of zero is symptomatic of a failed fit.
+  //
+  if (bestRChi2 < 0.001) {
+    cout<<"\t***ERROR***, FIT HAS FAILED for histo "<<gabscor->GetName()
+      <<" which has a reduced chi2="<<bestRChi2
+      <<" after "<<fitIter<<" iterations. "<<endl;
+  }
+
+  //
+  // check for large reduced chi2's
+  // above 10 is a plain error; between 5 and 10 is a warning
+  //
+  if (bestRChi2 > 5){
+    if (bestRChi2 > 10)
+      cout<<"\t***ERROR***,";
+    else
+      cout<<"\tWARNING,";
+
+    cout<<" fit for histo "<<gabscor->GetName()
+      <<" has a reduced chi2="<<bestRChi2
+      <<" after "<<fitIter<<" iterations"<<endl;
+  }
 }
 
 //______________________________________________________________________________
 void L2Creator::writeTextFileForCurrentAlgorithm() {
-	TString txtfilename = outputDir + era + "_L2Relative_" + ji->getAlias() + ".txt";
-	ofstream fout(txtfilename);
-	fout.setf(ios::right); 
-
-	unsigned int vector_size = 0;
-    if(l2l3) vector_size = vabscor_eta.size(); //For L2L3 Corrections Together
-    else vector_size = vrelcor_eta.size(); //For L2 & L3 Corrections Separate
-    for (unsigned int ieta=0;ieta<vector_size;ieta++) { 
-    	TGraph* grelcor;
-      	if(l2l3) grelcor = vabscor_eta[ieta]; //For L2L3 Corrections Together
-      	else grelcor = vrelcor_eta[ieta]; //For L2 & L3 Corrections Separate
-      	TF1* frelcor = (TF1*)grelcor->GetListOfFunctions()->Last();
-      	if(frelcor!=0) {
-      		if(ieta==0 || (ieta==1 && delphes))
-               fout<<"{1 JetEta 1 JetPt max(0.0001,"<<frelcor->GetTitle()<<") Correction L2Relative}"<<endl;
-               //fout<<"{1 JetEta 1 JetPt max(0.0001,"<<frelcor->GetExpFormula()<<") Correction L2Relative}"<<endl;
-      		double  etamin  = hl_jetpt.minimum(0,ieta);
-      		double  etamax  = hl_jetpt.maximum(0,ieta);
-      		double  ptmin = grelcor->GetX()[0];
-      		double  ptmax = grelcor->GetX()[grelcor->GetN()-1];
-      		fout<<setw(11)<<etamin
-      		<<setw(11)<<etamax
-        	    <<setw(11)<<(int)(frelcor->GetNpar()+2) //Number of parameters + 2 
-        	    <<setw(12)<<ptmin
-        	    <<setw(12)<<ptmax;
-        	    for(int p=0; p<frelcor->GetNpar(); p++) {
-        	    	fout<<setw(13)<<frelcor->GetParameter(p);
-        	    }
-        	    fout<<endl;
-        }
+  TString txtfilename = outputDir + "/" + era + "_L3Absolute_" + ji->getAlias() + ".txt";
```

Comment:
So it was just a choice? If so, why change anything at all? Nevertheless, I'm going to insist that you put this line back the way it was (i.e. use L2Relative). This code was meant to make L2 corrections and does not make L3 corrections, that's a difference process/code all together. Also this would "break" the current JEC workflow (we would need to rename each file that came out).